py/visdom/server.py

```diff
@@ -80,7 +81,8 @@ def extract_eid(args):
     "debug": "/dbg/" in __file__,
     "static_path": get_path('static'),
     "template_path": get_path('static'),
-    "compiled_template_cache": False
+    "compiled_template_cache": False,
+    "cookie_secret": "YOUR SECRET HERE"
```

Comment:
Try to retrieve this key from a `COOKIE_SECRET` file in the `'%s/.visdom/' % expanduser("~")` directory. If that file doesn't exist, have the user create the file the first time they try using a user/password near line 1094 (and then update the global `tornado_settings` there).